plugins/modules/python_requirements_info.py

```diff
@@ -165,7 +166,7 @@ def main():
             python_version_info=python_version_info,
             python_system_path=sys.path,
         )
-    pkg_dep_re = re.compile(r'(^[a-zA-Z][a-zA-Z0-9_-]+)(?:(==|[><]=?)([0-9.]+))?$')
+    pkg_dep_re = re.compile(r'(^[a-zA-Z][a-zA-Z0-9_-]+\[?([a-z][a-zA-Z0-9_ ]+){0,1}\]?)(?:(==|[><]=?)([0-9.]+))?$', flags=re.IGNORECASE)
```

Comment:
Shouldn't this be something like:
```suggestion
    pkg_dep_re = re.compile(r'(^[a-zA-Z][a-zA-Z0-9_-]+)(?:\[([a-z][a-zA-Z0-9_ ]+)\])?(?:(==|[><]=?)([0-9.]+))?$', flags=re.IGNORECASE)
```
Otherwise it will also accept stuff like `ansible[extra` or `ansibleextra]`.

Reply:
Let's check ;-)